src/test/java/uk/gov/hmcts/reform/sscs/service/HearingsMappingTest.java

```diff
@@ -0,0 +1,432 @@
+    @ParameterizedTest
+    @CsvSource(value = {"YES,YES", "NO,NO", "null,NO"}, nullValues = {"null"})
+    void shouldBeAdditionalSecurityFlag(YesNo additionalSecurityFlag, YesNo expected) {
+        // TODO Finish Test when method done
+        YesNo result = HearingsMapping.shouldBeAdditionalSecurityFlag(additionalSecurityFlag);
+
+        assertEquals(expected, result);
+    }
+
+    @DisplayName("shouldBeSensitiveFlag Parameterized Tests")
+    @ParameterizedTest
+    @CsvSource(value = {"YES,YES", "NO,NO", "null,NO"}, nullValues = {"null"})
+    void shouldBeSensitiveFlag(YesNo sensitiveFlag, YesNo expected) {
+        // TODO Finish Test when method done
+        YesNo result = HearingsMapping.shouldBeSensitiveFlag(sensitiveFlag);
+
+        assertEquals(expected, result);
+    }
+
+    @Test
+    @DisplayName("When case ID is given getCaseDeepLink returns the correct link")
+    void getCaseDeepLink() {
+        // TODO Finish Test when method done
+        String result = HearingsMapping.getCaseDeepLink(caseId);
+        String expected = String.format("%s/cases/case-details/%s", exUiUrl, caseId);
+
+        assertEquals(expected, result);
+    }
+
+    @Test
+    @DisplayName("When ... is given getCaseManagementLocationCode returns the correct EPIMS ID")
+    void getCaseManagementLocationCode() {
+        // TODO Finish Test when method done
+        CaseManagementLocation location = CaseManagementLocation.builder()
+                .baseLocation("Test Location")
+                .region("Test Region")
+                .build();
+        String result = HearingsMapping.getCaseManagementLocationCode(location);
+        String expected = null;
+
+        assertEquals(expected, result);
+    }
+
+    @Test
+    @DisplayName("When .. is given getFacilitiesRequired return the correct facilities Required")
+    void getFacilitiesRequired() {
+        // TODO Finish Test when method done
+        SscsCaseData caseData = SscsCaseData.builder().build();
+        List<CcdValue<String>> result = HearingsMapping.getFacilitiesRequired(caseData);
+        List<CcdValue<String>> expected = new ArrayList<>();
+
+        assertEquals(0, result.size());
+        assertEquals(expected, result);
+    }
+
+    @DisplayName("When case with valid DWP_RESPOND event and is auto-listable is given "
+            + "getHearingWindowRange returns a window starting within 1 month of the event's date")
+    @ParameterizedTest
+    @CsvSource(value = {
+        "DWP_RESPOND,YES,true",
+        "WITHDRAWN,YES,false",
+        "null,YES,false",
+        "DWP_RESPOND,NO,false",
+        "DWP_RESPOND,null,false",
+    }, nullValues = {"null"})
+    void getHearingWindowRange(EventType eventType, YesNo autoListFlag, boolean nonNullExpected) {
+        // TODO Finish Test when method done
+        List<Event> events = new ArrayList<>();
+        LocalDateTime testDateTime = LocalDateTime.now();
+        if (nonNull(eventType)) {
```

Comment:
Can you separate these out into two different tests - for the null and nonNull values?